app/controllers/accounts_controller.rb

```diff
@@ -132,4 +133,12 @@ def filtered_status_page(params)
       filtered_statuses.paginate_by_max_id(PAGE_SIZE, params[:max_id], params[:since_id]).to_a
     end
   end
+
+  def fields_selection
+    if signed_request_account.present? || !authorized_fetch_mode?
```

Comment:
this might be more clear with early returns?

```ruby
return unless authorized_fetch_mode?
return if signed_request_account.present?
%i(id type preferred_username inbox public_key endpoints)
```

Reply:
also i'm unclear what the point of returning a limited object is... mind explaining?

Reply:
Assuming that your goal is to prevent blocked instances from obtaining any data from yours, profile info is just as important as individual statuses. Bio text, header image, profile picture--all of that is quite personal. The attributes we do return here are the bare minimum, and technical in nature, to allow a two-way cryptography to be established.

Reply:
ah, right, a chicken and an egg sort of problem. we need to return the limited object so that we can authenticate to people who don't know who we are. hmm.

Reply:
Also no I would find two early returns harder to read in this case. "If we're signed in, or we don't require sign in, do this" reads more naturally to me.